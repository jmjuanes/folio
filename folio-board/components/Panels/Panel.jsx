import React from "react";
import classNames from "classnames";
// import {ChevronUpIcon} from "../icons/index.jsx";

export const Panel = props => {
    const panelWrapperClass = classNames(props.className, {
        "position:absolute z:5 select:none": true,
        "b:1 b:solid b:light-900": true,
        // "top:0 left:0 pl:4 pt:4": props.position === "top-left",
        // "top:0 right:0 pr:4 pt:4": props.position === "top-right",
        // "bottom:0 left:0 pb:4 pl:4": props.position === "bottom-left",
        // "bottom:0 right:0 pb:4 pr:4": props.position === "bottom-right",
        // "bottom:0 left:half pb:4": props.position === "bottom-center",
    });
    // TODO: we should move this to a classname
    const panelWrapperStyle = {
        // transform: props.position === "bottom-center" ? "translateX(-50%)" : null,
        ...props.style,
    };
    const panelContentClass = classNames({
        // "b-1 b-solid b-gray-900": true,
        "r:lg shadow:md items:center bg:white d:flex gap:2 p:2": true,
        // "flex:col": props.position === "top-left" || props.position === "top-right",
        "flex:col": props.direction === "col",
    });
    return (
        <div className={panelWrapperClass} style={panelWrapperStyle}>
            <div className={panelContentClass}>
                {props.children}
            </div>
        </div>
    );
};

Panel.defaultProps = {
    className: "",
    direction: "row",
};

export const PanelButton = props => {
    // TODO: missing height: 2.25rem height!!
    const classList = classNames({
        "items:center r:md d:flex text:lg p:2": true,
        "text:dark-700 bg:dark-800:hover text:white:hover cursor:pointer": !props.active && !props.disabled,
        "is-active bg:dark-800 text:white cursor:pointer": props.active && !props.disabled,
        "is-disabled text:dark-100 cursor:not-allowed o:60": !props.active && props.disabled,
    });
    return (
        <div className={classList} onClick={props.onClick}>
            {props.children}
        </div>
    );
};

PanelButton.defaultProps = {
    active: false,
    disabled: false,
};

export const PanelTextButton = props => {
    // TODO: missing height: 2.25rem height!!
    const classList = classNames({
        "d:flex flex:col w:16": true,
        "items:center r:md d:flex text:lg p:3": true,
        "text:dark-700 bg:dark-800:hover text:white:hover cursor:pointer": !props.active && !props.disabled,
        "is-active bg:dark-800 text:white cursor:pointer": props.active && !props.disabled,
        "is-disabled text:dark-100 cursor:not-allowed o:60": !props.active && props.disabled,
    });
    return (
        <div className={classList} onClick={props.onClick}>
            {props.children}
            <div className="mt:1 text:2xs">
                <strong>{props.text}</strong>
            </div>
        </div>
    );
};

PanelTextButton.defaultProps = {
    text: "",
    active: false,
    disabled: false,
};
