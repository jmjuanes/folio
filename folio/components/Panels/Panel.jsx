import React from "react";
import classNames from "classnames";
import {ChevronUpIcon} from "../icons/index.jsx";

export const Panel = props => {
    const panelWrapperClass = classNames({
        "position-absolute z-10": true,
        "top-0 left-0 pl-4 pt-4": props.position === "top-left",
        "top-0 right-0 pr-4 pt-4": props.position === "top-right",
        "bottom-0 left-0 pb-4 pl-4": props.position === "bottom-left",
        "bottom-0 right-0 pb-4 pr-4": props.position === "bottom-right",
        "bottom-0 left-half pb-4": props.position === "bottom-center",
    });
    // TODO: we should move this to a classname
    const panelWrapperStyle = {
        transform: props.position === "bottom-center" ? "translateX(-50%)" : null,
        ...props.style,
    };
    const panelContentClass = classNames({
        "b-1 b-solid b-gray-900 r-lg": true,
        "shadow-md items-center bg-white d-flex gap-2 p-2": true,
        "flex-col": props.position === "top-left" || props.position === "top-right",
    });
    return (
        <div className={panelWrapperClass} style={panelWrapperStyle}>
            <div className={panelContentClass}>
                {props.children}
            </div>
        </div>
    );
};

export const PanelButton = props => {
    // TODO: missing height: 2.25rem height!!
    const classList = classNames({
        "items-center r-md d-flex text-lg p-2": true,
        "text-gray-900 hover:bg-gray-900 hover:text-white cursor-pointer": !props.active && !props.disabled,
        "is-active bg-gray-900 text-white cursor-pointer": props.active && !props.disabled,
        "is-disabled text-gray-900 cursor-not-allowed o-60": !props.active && props.disabled,
    });
    return (
        <div className={classList} onClick={props.onClick}>
            {props.children}
        </div>
    );
};

PanelButton.defaultProps = {
    active: false,
    disabled: false,
};

export const PanelDropButton = props => {
    // TODO: missing height: 2.25rem height!!
    const classList = classNames({
        "d-flex r-md": true,
        "text-gray-900 hover:bg-gray-900 hover:text-white cursor-pointer": !props.active && !props.disabled,
        "is-active bg-gray-900 text-white cursor-pointer": props.active && !props.disabled,
        "is-disabled text-gray-900 cursor-not-allowed o-60": !props.active && props.disabled,
    });
    return (
        <div className={classList}>
            <div className="d-flex items-center text-lg p-2" onClick={props.onClick}>
                {props.children}
            </div>
            <div className="d-flex items-center text-sm py-2 pr-1" onClick={props.onDropClick}>
                <ChevronUpIcon />
            </div>
        </div>
    );
};

PanelDropButton.defaultProps = {
    active: false,
    onClick: null,
    onDropClick: null,
};
