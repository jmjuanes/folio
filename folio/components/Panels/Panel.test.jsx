import {create} from "react-test-renderer";

import {Panel, PanelButton} from "./Panel.jsx";

describe("Panel", () => {
    it("should render", () => {
        const component = create(<Panel position="top-left" />);

        expect(component.toJSON).toMatchSnapshot();
    });
});

describe("PanelButton", () => {
    it("should render", () => {
        const component = create(<Panel position="top-left" />);

        expect(component.toJSON).toMatchSnapshot();
    });

    it("should have the 'is-active' class when the active prop is 'true'", () => {
        const component = create(<PanelButton active={true} />);
        const root = component.root;

        expect(root.findByType("div").props.className).toEqual(
            expect.stringContaining("is-active"),
        );
    });

    it("should have the 'is-disabled' class when the disabled prop is 'true'", () => {
        const component = create(<PanelButton disabled={true} />);
        const root = component.root;

        expect(root.findByType("div").props.className).toEqual(
            expect.stringContaining("is-disabled"),
        );
    });
});
