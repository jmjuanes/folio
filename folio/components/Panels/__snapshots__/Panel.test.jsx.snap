// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel should render 1`] = `[Function]`;

exports[`PanelButton should render 1`] = `[Function]`;
